### ISSUE TITLE
BUILD: pandas 2.1.0 has no win32 wheels

### ISSUE THREAD

**Issue by Zac-HD** [pandas-dev/pandas#54979]
It looks like the root cause is that there are [no win32 wheels published for Pandas 2.1.0](https://pypi.org/project/pandas/2.1.0/#files) (or [the prerelease](https://pypi.org/project/pandas/2.0.0rc1/#files)), although [there were for 2.0.3](https://pypi.org/project/pandas/2.0.3/#files) and for [the 2.0.0rc1](https://pypi.org/project/pandas/2.0.3/#files) release.  

I don't mind at all if you've decided to stop building wheels for this rather outdated platform, but if so it would be good to say so in the [changelog](https://pandas.pydata.org/docs/whatsnew/v2.1.0.html) and perhaps [install instructions](https://pandas.pydata.org/pandas-docs/stable/getting_started/install.html) since this has just changed!

### Installation check

- [X] I have read the [installation guide](https://pandas.pydata.org/pandas-docs/stable/getting_started/install.html#installing-pandas).

### Platform

```
Operating System
  Microsoft Windows Server 2022
  10.0.20[3](https://github.com/HypothesisWorks/hypothesis/actions/runs/6063159411/job/16450163482?pr=3728#step:1:3)[4](https://github.com/HypothesisWorks/hypothesis/actions/runs/6063159411/job/16450163482?pr=3728#step:1:4)[8](https://github.com/HypothesisWorks/hypothesis/actions/runs/6063159411/job/16450163482?pr=3728#step:1:9)
  Datacenter
Runner Image
  Image: windows-2022
  Version: 20230820.1.0
  Included Software: https://github.com/actions/runner-images/blob/win22/20230820.1/images/win/Windows2022-Readme.md
  Image Release: https://github.com/actions/runner-images/releases/tag/win22%2F20230820.1
```

### Installation Method

pip install

### pandas Version

2.1.0

### Python Version

3.11.4

### Installation Logs

<details>

https://github.com/HypothesisWorks/hypothesis/actions/runs/6063159411/job/16450163482?pr=3728#step:5:141

```
Collecting pandas==2.1.0 (from -r requirements/coverage.txt (line 47))
  Downloading pandas-2.1.0.tar.gz (4.3 MB)
     ---------------------------------------- 4.3/4.3 MB 22.7 MB/s eta 0:00:00
  Installing build dependencies: started
  Installing build dependencies: finished with status 'done'
  Getting requirements to build wheel: started
  Getting requirements to build wheel: finished with status 'done'
  Preparing metadata (pyproject.toml): started
  Preparing metadata (pyproject.toml): finished with status 'error'
  error: subprocess-exited-with-error
  
  Preparing metadata (pyproject.toml) did not run successfully.
  exit code: 1
  
  [22 lines of output]
  + meson setup C:\Users\runneradmin\AppData\Local\Temp\pip-install-cbm_veh1\pandas_8bb52d05edfc4b298f3afe3141e1f01c C:\Users\runneradmin\AppData\Local\Temp\pip-install-cbm_veh1\pandas_8bb52d05edfc4b298f3afe3141e1f01c\.mesonpy-y6wsmxqe\build -Dbuildtype=release -Db_ndebug=if-release -Db_vscrt=md --vsenv --native-file=C:\Users\runneradmin\AppData\Local\Temp\pip-install-cbm_veh1\pandas_8bb52d05edfc4b298f3afe3141e1f01c\.mesonpy-y6wsmxqe\build\meson-python-native-file.ini
  The Meson build system
  Version: 1.0.1
  Source dir: C:\Users\runneradmin\AppData\Local\Temp\pip-install-cbm_veh1\pandas_8bb52d05edfc4b298f3afe3141e1f01c
  Build dir: C:\Users\runneradmin\AppData\Local\Temp\pip-install-cbm_veh1\pandas_8bb52d05edfc4b298f3afe3141e1f01c\.mesonpy-y6wsmxqe\build
  Build type: native build
  Project name: pandas
  Project version: 2.1.0
  Activating VS 17.7.1
  C compiler for the host machine: cl (msvc 19.36.32538 "Microsoft (R) C/C++ Optimizing Compiler Version 19.36.32538 for x64")
  C linker for the host machine: link link 14.36.32538.0
  C++ compiler for the host machine: cl (msvc 19.36.32538 "Microsoft (R) C/C++ Optimizing Compiler Version 19.36.32538 for x64")
  C++ linker for the host machine: link link 14.36.32538.0
  Cython compiler for the host machine: cython (cython 0.29.36)
  Host machine cpu family: x[86](https://github.com/HypothesisWorks/hypothesis/actions/runs/6063159411/job/16450163482?pr=3728#step:5:87)_64
  Host machine cpu: x86_64
  Program python found: YES (C:\hostedtoolcache\windows\Python\3.11.4\x86\python.exe)
  Need python for 64-bit, but found 32-bit
  
  ..\..\pandas\_libs\tslibs\meson.build:23:7: ERROR: Python dependency not found
  
  A full log can be found at C:\Users\runneradmin\AppData\Local\Temp\pip-install-cbm_veh1\pandas_8bb52d05edfc4b2[98](https://github.com/HypothesisWorks/hypothesis/actions/runs/6063159411/job/16450163482?pr=3728#step:5:99)f3afe3141e1f01c\.mesonpy-y6wsmxqe\build\meson-logs\meson-log.txt
  [end of output]
```

</details>

**Comment by phofl** [comment]
cc @lithomas1

**Comment by lithomas1** [comment]
This was intentional. IIRC, around 1% of people were downloading the win32 wheel, and cross-compiling it is not easy so I didn't put them up.

**Comment by Raghav-Bell** [comment]
Please assign this issue to me.

**Comment by phofl** [comment]
@lithomas1 do you want to add them back in? I am ok with removing them as long as it was intentional

**Comment by lithomas1** [comment]
> @lithomas1 do you want to add them back in? I am ok with removing them as long as it was intentional

Yeah, this was intentional. The current issue is to document supported platforms I guess. 

Below Python version support here(https://pandas.pydata.org/docs/getting_started/install.html#python-version-support), we should write a list of supported platforms.

Off of the top of my head, that's 
x86_64 and aarch64 on manylinux2014 platforms (manylinux 2_17 on linux), 
macOS > OSX 10.9 for Intel x86_64, macOS > 11 for M1,
and x86_64 on Windows.

If someone wants to note this down, that'd be great

**Comment by chrisoro** [comment]
@lithomas1 I understand your reluctance to provide wheels for a minority group. Could you please share instructions on how to build it myself?
I've attempted to build a Win32 wheel for the newer releases, but I'm encountering difficulties getting meson to build it properly.

**Comment by mroeschke** [comment]
Since we already released 2.2, there will be no more changes to 2.1 wheels so closing
